pkg/storage/tsdb/config.go

```diff
@@ -48,31 +48,38 @@ type Config struct {
 // DurationList is the block ranges for a tsdb
 type DurationList []time.Duration
 
-// String implements the flag.Var interface
-func (d DurationList) String() string { return "RangeList is the block ranges for a tsdb" }
-
-// Set implements the flag.Var interface
-func (d DurationList) Set(s string) error {
-	blocks := strings.Split(s, ",")
-	d = make([]time.Duration, 0, len(blocks)) // flag.Parse may be called twice, so overwrite instead of append
-	for _, blk := range blocks {
-		t, err := time.ParseDuration(blk)
+// String implements the flag.Value interface
+func (d *DurationList) String() string {
+	values := make([]string, 0, len(*d))
+	for _, v := range *d {
+		values = append(values, v.String())
+	}
+
+	return strings.Join(values, ",")
+}
+
+// Set implements the flag.Value interface
+func (d *DurationList) Set(s string) error {
+	values := strings.Split(s, ",")
+	*d = make([]time.Duration, 0, len(values)) // flag.Parse may be called twice, so overwrite instead of append
```

Comment:
This is tricky. flag package supports calling same parameter multiple times, in which case it will also call `Set` method multiple times. More flag-compatible way of using it would therefore be: `-compactor.block-ranges=2h -compactor.block-ranges=48h`. It's ok if we want to support list flag, but we should probably still append ranges from multiple `Set` calls.

In either case, should these ranges be validated for being in correct order, before they are used?

Reply:
This way done by @thorfour. I've just fixed it because didn't work the CLI flags parsing (was passed by copy and not reference).

@thorfour What's the use case because you did this? I'm missing some context to safely change it.

Reply:
I guess I was hedging a little too much against obviousness of behaviour. I thought if you had multiple flags with duration, it wasn't obvious in which order they are applied. But looking back on it now, it's pretty obvious they'll be sorted by time. I think it's probably a good change to get rid of that, and just use multiple flags if desired.

Reply:
The clearing of the duration list was because the flag parse function is called twice from the main package, which was resulting in a double list being created, despite the user only submitting a single list.

Reply:
I like single comma-separated parameter option as well, but think that multiple parameters should be supported, and "append" is a correct option to use here, instead of overwrite.

About sorting -- where does that happen? I haven't found validation nor sorting (but haven't looked that hard)

Reply:
> The clearing of the duration list was because the flag parse function is called twice from the main package, which was resulting in a double list being created, despite the user only submitting a single list.

Oh really? That would actually explain one weird thing in memberlist implementation I was seeing and never connected to flags being parsed twice. We should fix that, if possible.

Reply:
Sorry, was kind of talking out loud. If we moved away from a comma separated parameter list and just did a single time duration flag that could be passed multiple times, we'd want to sort all of those values so we didn't end up with block ranges that made no sense. 

I think you're right in that "append" is a correct behaviour if given two flags, it just gets weird when `flag.Parse` is called twice from main and there's only one flag.

Reply:
I will send PR that fixes flag.Parse() being called twice on same config soon.

Reply:
This discussion is very interesting. I've seen the CLI flags are parsed twice in the `main.go` but I'm not sure about the reason (I agree should be fixed, not sure about the implications). In any case looks out of the scope of this PR, so I would keep it for a separate PR.

Reply:
The only reason for double-parsing is that we need -config.file option to parse config file, and then reparse command line flags so that they take precendence. Easy to fix. I'll send separate PR. It came up here, because this `Set` method is using incorrect semantics (replacing instead of appending flags).

Reply:
I've sent #1997 now.

Reply:
Now that #1997 is merged (with subsequent fix as well), it would make sense to use `*d = append(*d, values...)` here, to make it work like standard flag. What do you think?

Reply:
Ah, I guess I'm too late. I can send separate PR later. :) Great to see this merged!

Reply:
I didn't wait to merge this PR before addressing this because it's not a regression introduced by this PR. I would be glad if you could take care of that 🙏